src/services/Tx.ts

```diff
@@ -0,0 +1,38 @@
+import { ApiPromise } from '@polkadot/api'
+import { SubmittableExtrinsic } from '@polkadot/api/types'
+import BN from 'bn.js'
+import { Query } from './Query'
+
+type Itx = {
+  createPool(
+    api: ApiPromise,
+    address: string,
+    firstAssetId: string,
+    firstAssetAmount: BN,
+    secondAssetId: string,
+    secondAssetAmount: BN
+  ): Promise<void>
+}
+
+const signTx = async (tx: SubmittableExtrinsic<'promise'>, address: string, nonce: string) => {
+  await tx.signAndSend(address, { nonce })
+}
+
+const createPool = async (
+  api: ApiPromise,
+  address: string,
+  firstAssetId: string,
+  firstAssetAmount: BN,
+  secondAssetId: string,
+  secondAssetAmount: BN
+): Promise<void> => {
+  signTx(
+    api.tx.xyk.createPool(firstAssetId, firstAssetAmount, secondAssetId, secondAssetAmount),
+    address,
+    await Query.getCurrentNonce(api, address)
```

Comment:
Add proper nonce management:
- make nonce parameter optional;
- if not provided, pick the nonce stored in memory;
- if no nonce stored in memory, query nonce from chain.

Also the nonce stored in memory should be updated in case of tx failure.